src/context/filters/filtersSlice.ts

```diff
@@ -23,20 +25,30 @@ export const slice = createSlice({
       ...state,
       layers: layers.add(payload),
     }),
+
     removeLayer: (
       { layers, ...state },
       { payload }: PayloadAction<LayerType>,
     ) => ({
       ...state,
       layers: layers.delete(payload),
     }),
+
+    updateDate: ({ layers, ...state }, { payload }: PayloadAction<Date>) => ({
+      ...state,
+      // 'layers' is getting converted into a object instead of Set<LayerType>
```

Comment:
This seems unsustainable - we don't want to need to specifically call out parts of state in unrelated reducers all over (and maintain that as we add to state). Generally speaking, we're going to run into issues if we mix and match Immutable.js with regular JS like this.

[Here's some good reading on redux with TypeScript](https://redux.js.org/recipes/using-immutablejs-with-redux)

If we want to use Immutable.js for state, I think we need to use it completely (which means foregoing de-structuring), and having the whole of state be a `Map` rather than a plain JS object. Then a reducer would involve `state.set` or `state.withMutation` or something like that.

- or - 

We could just go back to regular JS objects for state.

I confess that I do generally like Immutable structures a lot, but if you don't want to deal with the complexity of that, that's fine. But I don't think we should be mixing & matching JS objects & Immutable ones like this.